Source/Core/InputCommon/ControllerInterface/Device.cpp

```diff
@@ -189,5 +189,11 @@ Device::Output* DeviceContainer::FindOutput(const std::string& name, const Devic
 {
   return def_dev->FindOutput(name);
 }
+
+bool DeviceContainer::HasConnectedDevice(const DeviceQualifier& qualifier) const
+{
+  const auto device = FindDevice(qualifier);
+  return device.get() && device->IsValid();
```

Comment:
You don't need to call `get()` on `std::shared_ptr` and `std::unique_ptr` to check pointer nullity; they have comparison operators implemented for checking against `nullptr`, so you can just do `device != nullptr`